### DISCUSSION TITLE
How to use Prisma in Nextjs13 AppDir from Client-Side

### DISCUSSION THREAD

**Discussion by TheLexoPlexx** [prisma/prisma#19020] [Q&A] [answered]
Hi,

this may be a really stupid questions, I am probably overlooking something but here's the thing:

I've got a project with lots of models and most of them are connected with each other.

The thing is: I am manipulating that data through forms and in order to perform basically anything (such as simple CRUD), I'd have to write an API Endpoint for every single task of CRUD for quite a few models, this creates lots of boilerplate code which is not just tedious, but also loses the beautiful type-safety of using Prisma with Typescript and will probably be prone to errors, let alone changes in the future.

Is there any useful way of utilitzing the sweet syntactical sugar of prisma in client-side somehow? Maybe sort of like a wrapper or generator? I'm not a security expert, but wouldn't it be sufficient to create a wrapper, which signs a request with a nonce and performs these tasks? Probably not, otherwise someone would've already done it, is my guess.

Or should i just yoink it and write upsert-endpoints for all the models?

I did have a look at trpc but I didn't get it at first glance. (It also officially does not support Next App-Dir yet. (https://github.com/trpc/trpc/discussions/3185)

If my point is not clear enough, I'll try to explain it with different words or code. Thank you for any ideas in advance.

(I asked the same question over on nextjs 26 days ago, (https://github.com/vercel/next.js/discussions/47876))

**Comment by nurul3101** [comment]
Hey @TheLexoPlexx 👋 

We have a feature request to allow using Prisma on the Client side.
- https://github.com/prisma/prisma/issues/4992

I believe you want to achieve something similar, right?

**Reply by TheLexoPlexx** [reply]
Yes, that is basically what I want to achieve.

**Comment by TheLexoPlexx** [comment, marked as answer]
This problem was solved in Next.Js 13.4 with Server Actions, can be found over here: https://github.com/vercel/next.js/discussions/47876#discussioncomment-5823411

**Reply by nurul3101** [reply]
Thanks for sharing the link 👍
